src/index.ts

```diff
@@ -75,21 +78,61 @@ export class MyMCP extends McpAgent<Env> {
 	initialState: State = {
 		projectId: undefined,
 		orgId: undefined,
+		distinctId: undefined,
 	};
 
-	_cache: MemoryCache<State> | undefined;
+	_cache: DurableObjectCache<State> | undefined;
 
 	get requestProperties() {
 		return this.props as RequestProperties;
 	}
 
 	get cache() {
 		if (!this._cache) {
-			this._cache = new MemoryCache<State>(this.requestProperties.userHash);
+			this._cache = new DurableObjectCache<State>(this.requestProperties.userHash, this.ctx.storage);
 		}
+		
 		return this._cache;
 	}
 
+	async getDistinctId() {
+		let _distinctId = await this.cache.get("distinctId");
+
+		if (!_distinctId) {
+			const user = await getUser(this.requestProperties.apiToken);
+			await this.cache.set("distinctId", user.distinctId);
+			_distinctId = user.distinctId;
+		}
+
+		return _distinctId;
+	}
+
+	async trackEvent(event: string, properties: Record<string, any> = {}) {
+		const distinctId = await this.getDistinctId();
+
+		const client = getPostHogClient();
+
+		client.capture({ distinctId, event, properties });
```

Comment:
i think this should be wrapped in a try catch otherwise `getDistinctId` throws and the whole tool call will fail right